### PR TITLE
cvss: bugs, tests

### PR DESCRIPTION
**Pull request by hdonnay**
This fixes some bugs and moves the test fixtures to external files (something that was bugging me).

Closes: #1382

### CODE REVIEW

File: toolkit/types/cvss/cvss_v2_parse.rl [hunk 1]
```diff
@@ -53,9 +53,11 @@ cr  = ( "CR"  >metric_start %mark_metric) ":" ( ("L"|"M"|"H"|"ND")           >va
 ir  = ( "IR"  >metric_start %mark_metric) ":" ( ("L"|"M"|"H"|"ND")           >value_start %mark_value);
 ar  = ( "AR"  >metric_start %mark_metric) ":" ( ("L"|"M"|"H"|"ND")           >value_start %mark_value);
 
-metric = ( av | ac | au | c | i | a | e | rl | rc | cdp | td | cr | ir | ar );
+base = av "/" ac "/" au "/" c "/" i "/" a;
+temporal = e "/" rl "/" rc;
+environmental = cdp "/" td "/" cr "/" ir "/" ar;
```
Comment (RTann) [inline toolkit/types/cvss/cvss_v2_parse.rl:58]: I'm not as familiar with these. Let's say `E` is set. Does that mean `RL` and `RC` both must also be set?

File: toolkit/types/cvss/cvss_v4.go [hunk 2]
```diff
@@ -62,7 +62,7 @@ func (v *V4) getString(m V4Metric) (string, error) {
 // GetScore implements [Vector].
 func (v *V4) getScore(m V4Metric) byte {
 	b := v.mv[int(m)]
-	if m >= V4ExploitMaturity && b == 0 {
+	if m >= V4ExploitMaturity && (b == 0 /* not present */ || b == 'X' /* not defined */) {
```
Comment (RTann) [inline toolkit/types/cvss/cvss_v4.go:65]: Are you able to show me where in the spec it says this? The 4.0 spec does not seem to show me a nice formula that v3.1 had

File: toolkit/types/cvss/cvss_v2_parse.rl (hunk 1 above)
Comment (hdonnay) [reply toolkit/types/cvss/cvss_v2_parse.rl:58]: I think so, yes. The v2 spec is (IMO) unclear on this, but seems to say that if you use a metric group, all the metrics must be filled out.

File: toolkit/types/cvss/cvss_v4.go (hunk 2 above)
Comment (hdonnay) [reply toolkit/types/cvss/cvss_v4.go:65]: Where it says what? That these have specific defaults? That's documented under the relevant tables in the [v4 scoring section](https://www.first.org/cvss/v4.0/specification-document#CVSS-v4-0-Scoring-using-MacroVectors-and-Interpolation).

**Review by crozzy — APPROVED** [review]
my ragel understanding is a WIP but I think the changes make sense

**Pull request merged**